packages/cli/README.md

```diff
@@ -127,3 +127,615 @@ jobs:
       - run: npm ci
       - run: npx code-pushup autorun --upload.apiKey=${{ secrets.PORTAL_API_KEY }}
 ```
+
+## Custom Plugins
+
+One of the main features of Code PushUp is the ability to write custom plugins to track your own metrics.
+It enables you to implement neraly any kind of metric you want to track with minimum effort.
+In this section we will go through the steps to create a custom plugin and integrate it in your project.
+
+### Set up
+
+To start crafting custom plugins you need a minimum `code-pushup.config.ts` file including the `persist` and `plugins`
+property.
+
+**config and plugin template**
+
+```typescript
+// code-pushup.config.ts
+import { AuditOutputs, PluginConfig } from '@code-pushup/models';
+
+type Options = {};
+async function create(options: Options): PluginConfig {
+  const audit = {
+    slug: 'my-audit',
+    title: 'My audit',
+    description: 'My custom audit.',
+  };
+  return {
+    slug: 'my-plugin',
+    title: 'My plugin',
+    icon: 'javascript', // icon name from [vscode-material-icon-theme](https://github.com/PKief/vscode-material-icon-theme/tree/main/icons)
+    description: 'My custom plugin.',
+    audits: [audit],
+    runner: (): AuditOutputs => {
+      // Dummy audit output
+      const auditOutput: AuditOutput = {
+        slug: audit.slug,
+        score: 0,
+        value: 0,
+      };
+      console.log('In plugin runner');
+      // return dummy data of type `AuditOutputs` as plugin result
+      return [auditOutput];
+    },
+  };
+}
+
+export default {
+  persist: {
+    outputDir: '.code-pushup',
+  },
+  plugins: [await create()],
+  categories: [],
+};
+```
+
+Execute the CLI with `npx code-pushup collect --no-progress` and you should the following output:
+
+**stdout of plugin template**
```

Comment:
As discussed, it would be valuable to show a realistic output and perhaps mention as a `detail` what the example above would produce.

Additionally, it would be helpful to list what one can expect in the summary.